### PR TITLE
Generate git revision hash into geos_c.h

### PR DESCRIPTION
**Pull request by pramsey**
instead of a separate .h file

### CODE REVIEW

**Comment by mloskot** [comment]
LGTM. Thx!

File: capi/CMakeLists.txt [hunk 1]
```diff
@@ -9,18 +9,20 @@
 # See the COPYING file for more information.
 ################################################################################
 
-if(GEOS_BUILD_FROM_GIT)
-  add_custom_target(geos_revision_header
```
Comment (mloskot) [inline capi/CMakeLists.txt]: Replacing the target w/ plain command changes the procedure a bit
AFAIR, previous version re-generated the info on every build, not every re-configuration.

Comment (dbaston) [reply capi/CMakeLists.txt]: I think it should be regenerated on each build, or just left out. A potentially incorrect hash seems to have more confusion potential than no hash.

Comment (pramsey) [reply capi/CMakeLists.txt]: Well, generating every build has it's own downside, which Sandro alluded to on channel, which is the regen can end up kicking off large scale rebuilds over the code when not really necessary. I'm inclined to remove the whole thing.

Comment (dbaston) [reply capi/CMakeLists.txt]: > I'm inclined to remove the whole thing.

Agreed

Comment (mloskot) [reply capi/CMakeLists.txt]: >  regen can end up kicking off large scale rebuilds over the code 

It shouldn't, if `geos_revision.h` promises to stick to `geos_ts.cpp` only.

> I'm inclined to remove the whole thing.

Sweet

Comment (pramsey) [reply capi/CMakeLists.txt]: Harsher approach at #174

**Pull request merged**